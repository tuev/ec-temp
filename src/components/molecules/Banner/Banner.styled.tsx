/**
 *
 * Styled for Banner
 *
 */

import styled from 'styled-components'
import { prop } from 'styled-tools'

export const BannerWrapper = styled.div<{ background: string }>`
  height: calc(100vh - 80px);
  width: 100vw;
  background-image: url(${prop('background')});
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  padding: 64px 128px;
`
