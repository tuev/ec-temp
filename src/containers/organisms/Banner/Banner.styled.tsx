/**
 *
 * Styled for Banner
 *
 */

import styled from 'styled-components'
import { prop } from 'styled-tools'

export const BannerWrapper = styled.div<{ background: string }>`
  height: 440px;
  width: 100%;
  background-image: url(${prop('background')});
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
`
