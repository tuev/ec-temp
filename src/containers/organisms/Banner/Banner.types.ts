export interface IBanner {
  background?: string
  title: string
  subtitle?: string
  action?: React.MouseEventHandler<HTMLElement>
}

export type BannerProps = IBanner
