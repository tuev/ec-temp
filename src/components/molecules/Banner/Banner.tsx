/**
 *
 * Banner
 *
 */

import React, { FC, useCallback } from 'react'
import Wrapper from 'components/atoms/Wrapper'
import Typography from 'components/atoms/Typography'
import Button from 'components/atoms/Button'
import { BannerWrapper } from './Banner.styled'
import { BannerProps } from './Banner.types'
import mockImage from './mock/mock.png'
import { COLOR } from 'theme/colors'

const Banner: FC<BannerProps> = (props: BannerProps) => {
  const {
    title,
    subtitle = '',
    action,
    background = mockImage,
  }: BannerProps = props
  const handleClick = useCallback(e => action && action(e), [action])

  return (
    <BannerWrapper background={background}>
      <Wrapper pb="1rem" maxWidth="50%">
        <Typography customvariant="header1" customcolor="white">
          {title}
        </Typography>
      </Wrapper>
      <Wrapper pb="2rem" maxWidth="50%">
        <Typography customvariant="header2" customcolor="white">
          {subtitle}
        </Typography>
      </Wrapper>
      <Button onClick={handleClick} variant="outlined" data-testid="banner-btn">
        Shop now
      </Button>
    </BannerWrapper>
  )
}

export default Banner
