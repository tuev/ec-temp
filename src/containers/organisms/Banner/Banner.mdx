# Stories mdx for Banner

import { Meta, Story } from '@storybook/addon-docs/blocks'
import * as stories from './Banner.stories.tsx'
import Banner from './Banner.tsx'

<Story name="basic" />