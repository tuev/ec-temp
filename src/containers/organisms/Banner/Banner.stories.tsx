/**
 *
 *  Stories for Banner
 *
 */

import React from 'react'
import { action } from '@storybook/addon-actions'
import Banner from './Banner'

export default {
  component: Banner,
  includeStories: ['basic'],
  title: 'Banner',
}

export const basic = () => (
  <div>
    <Banner
      title="Background and Development"
      subtitle="This is your background subtitle"
    />
  </div>
)
