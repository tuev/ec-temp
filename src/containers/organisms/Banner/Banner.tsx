/**
 *
 * Banner
 *
 */

import React, { FC, useCallback } from 'react'
import Wrapper from 'components/atoms/Wrapper'
import Typography from 'components/atoms/Typography'
import Button from 'components/atoms/Button'
import { BannerWrapper } from './Banner.styled'
import { BannerProps } from './Banner.types'
import mockImage from './mock/mock.png'

const Banner: FC<BannerProps> = (props: BannerProps) => {
  const {
    title = 'Background and Development',
    subtitle = 'This is your background subtitle',
    action,
    background = mockImage,
  }: BannerProps = props
  const handleClick = useCallback(e => action && action(e), [action])

  return (
    <BannerWrapper background={background}>
      <Wrapper p="62px 128px">
        <Wrapper pb="1rem" maxWidth="50%">
          <Typography customvariant="header1" customcolor="white">
            {title}
          </Typography>
        </Wrapper>
        <Wrapper pb="2rem" maxWidth="50%">
          <Typography customvariant="header2" customcolor="white">
            {subtitle}
          </Typography>
        </Wrapper>
        <Button
          onClick={handleClick}
          variant="outlined"
          data-testid="banner-btn"
        >
          Shop now
        </Button>
      </Wrapper>
    </BannerWrapper>
  )
}

export default Banner
