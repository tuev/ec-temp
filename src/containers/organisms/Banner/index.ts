/**
 *
 * Index for Banner
 *
 */

import Banner from './Banner'

export default Banner
