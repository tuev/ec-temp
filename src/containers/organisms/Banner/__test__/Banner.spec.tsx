/**
 *
 * Tests for Banner
 *
 */

import React from 'react'

import { render, fireEvent } from '@testing-library/react'
import Banner from '..'

describe('App Banner', () => {
  it('Test get default Banner no params ', () => {
    // test snapshot
    const wrapper = render(<Banner title="banner title" />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Test get default Banner no params with action', () => {
    // test snapshot
    const actionHandler = jest
      .fn()
      .mockImplementation((e) => console.log('click banner'))
    const wrapper = render(
      <Banner action={actionHandler} title="banner title" />
    )
    expect(wrapper).toMatchSnapshot()

    const bannerBtn = wrapper.getByTestId('banner-btn')
    fireEvent.click(bannerBtn)
    expect(actionHandler).toHaveBeenCalled()
  })
})
