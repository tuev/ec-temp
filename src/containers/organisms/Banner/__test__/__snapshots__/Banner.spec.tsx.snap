// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Banner Test get default Banner no params  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  padding: 62px 128px;
}

.c2 {
  padding-bottom: 1rem;
  max-width: 50%;
}

.c4 {
  padding-bottom: 2rem;
  max-width: 50%;
}

.c3 {
  font-size: 48px;
  line-height: 1.5;
  font-family: 'Playfair Display';
  font-weight: bold;
  color: #fff;
}

.c5 {
  font-size: 24px;
  line-height: 1.5;
  font-family: 'Montserrat Alternates';
  font-weight: bold;
  color: #fff;
}

.c6 {
  min-width: 163px !important;
  min-height: 48px !important;
  font-size: 14px;
  border-radius: 32px;
}

.c0 {
  height: 440px;
  width: 100%;
  background-image: url(mock.png);
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <p
              class="MuiTypography-root c3 MuiTypography-body1"
              customcolor="white"
              customvariant="header1"
            >
              banner title
            </p>
          </div>
          <div
            class="c4"
          >
            <p
              class="MuiTypography-root c5 MuiTypography-body1"
              customcolor="white"
              customvariant="header2"
            >
              This is your background subtitle
            </p>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined c6 MuiButton-outlinedPrimary"
            data-testid="banner-btn"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Shop now
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxhCb eEEZGs"
    >
      <div
        class="sc-AxjAm wQNUV"
      >
        <div
          class="sc-AxjAm QCVyp"
        >
          <p
            class="MuiTypography-root sc-AxirZ hASgob MuiTypography-body1"
            customcolor="white"
            customvariant="header1"
          >
            banner title
          </p>
        </div>
        <div
          class="sc-AxjAm cuayUK"
        >
          <p
            class="MuiTypography-root sc-AxirZ gMnndD MuiTypography-body1"
            customcolor="white"
            customvariant="header2"
          >
            This is your background subtitle
          </p>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined sc-AxiKw inYPse MuiButton-outlinedPrimary"
          data-testid="banner-btn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Shop now
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App Banner Test get default Banner no params with action 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c1 {
  padding: 62px 128px;
}

.c2 {
  padding-bottom: 1rem;
  max-width: 50%;
}

.c4 {
  padding-bottom: 2rem;
  max-width: 50%;
}

.c3 {
  font-size: 48px;
  line-height: 1.5;
  font-family: 'Playfair Display';
  font-weight: bold;
  color: #fff;
}

.c5 {
  font-size: 24px;
  line-height: 1.5;
  font-family: 'Montserrat Alternates';
  font-weight: bold;
  color: #fff;
}

.c6 {
  min-width: 163px !important;
  min-height: 48px !important;
  font-size: 14px;
  border-radius: 32px;
}

.c0 {
  height: 440px;
  width: 100%;
  background-image: url(mock.png);
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
}

<div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <p
              class="MuiTypography-root c3 MuiTypography-body1"
              customcolor="white"
              customvariant="header1"
            >
              banner title
            </p>
          </div>
          <div
            class="c4"
          >
            <p
              class="MuiTypography-root c5 MuiTypography-body1"
              customcolor="white"
              customvariant="header2"
            >
              This is your background subtitle
            </p>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined c6 MuiButton-outlinedPrimary"
            data-testid="banner-btn"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Shop now
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxhCb eEEZGs"
    >
      <div
        class="sc-AxjAm wQNUV"
      >
        <div
          class="sc-AxjAm QCVyp"
        >
          <p
            class="MuiTypography-root sc-AxirZ hASgob MuiTypography-body1"
            customcolor="white"
            customvariant="header1"
          >
            banner title
          </p>
        </div>
        <div
          class="sc-AxjAm cuayUK"
        >
          <p
            class="MuiTypography-root sc-AxirZ gMnndD MuiTypography-body1"
            customcolor="white"
            customvariant="header2"
          >
            This is your background subtitle
          </p>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined sc-AxiKw inYPse MuiButton-outlinedPrimary"
          data-testid="banner-btn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Shop now
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
